src/test/java/com/example/task_2/hibernate_service_test/HibernateUserServiceTest.java

```diff
@@ -0,0 +1,155 @@
+    }
+
+    @Test
+    public void deleteTicketById_RuntimeException() {
+        doThrow(new RuntimeException("Some message")).when(userDAO).deleteUserById(1L);
+
+        Exception e = assertThrows(RuntimeException.class,
+                () -> userService.deleteTicketById(1L));
+
+        assertEquals("Some message", e.getMessage());
+        verify(userDAO, times(1)).deleteUserById(1L);
+    }
+
+    @Test
+    public void getUserById() {
+        User user = new User(1L);
+
+        when(userDAO.getUserById(1L)).thenReturn(user);
+        User userFromTest = userService.getUserById(1L);
+
+        assertEquals(1L, userFromTest.getId());
+        verify(userDAO, times(1)).getUserById(1L);
+    }
+
+    @Test
+    public void getUserById_RuntimeException() {
+        when(userDAO.getUserById(1L)).thenThrow(new RuntimeException("Error message"));
+        Exception e = assertThrows(RuntimeException.class,
+                () -> userService.getUserById(1L));
+
+        assertEquals("Error message", e.getMessage());
+        verify(userDAO, times(1)).getUserById(1L);
+    }
+
+    @Test
+    public void getUserTickets() {
+        List<Ticket> tickets = List.of(
+                new Ticket(1L),
+                new Ticket(2L)
+        );
+
+        when(userDAO.getUserTickets(1L)).thenReturn(tickets);
+        List<Ticket> ticketsFromTest = userService.getUserTickets(1L);
+
+        assertEquals(2, ticketsFromTest.size());
+        verify(userDAO, times(1)).getUserTickets(1L);
+    }
+
+    @Test
+    public void getUserTickets_RuntimeException() {
+        when(userDAO.getUserTickets(1L)).thenThrow(new RuntimeException("Failed to fetch tickets!"));
+        Exception e = assertThrows(RuntimeException.class,
+                () -> userService.getUserTickets(1L));
+
+        assertEquals("Failed to fetch tickets!", e.getMessage());
+        verify(userDAO, times(1)).getUserTickets(1L);
+    }
+
+    @Test
+    public void saveUser() {
+        User userToStore = new User(1L);
+
+        doNothing().when(userDAO).saveUser(userToStore);
+        userService.saveUser(userToStore);
+
+        verify(userDAO, times(1)).saveUser(userToStore);
+    }
+
+    @Test
+    public void saveUser_RuntimeException() {
+        User u = new User(1L);
+
+        doThrow(new RuntimeException("Failed to store user!")).when(userDAO).saveUser(u);
+        Exception e = assertThrows(RuntimeException.class,
+                () -> userService.saveUser(u));
+
+        assertEquals("Failed to store user!", e.getMessage());
+        verify(userDAO, times(1)).saveUser(u);
+    }
+
+    @Test
+    public void updateUserNameAndCreateTicket() {
+        Ticket ticketToStore = new Ticket(20L);
+
+        doNothing().when(userDAO).updateUserNameAndSaveTicket(1L, "new Name", ticketToStore);
+        userService.updateUserNameAndCreateTicket(1L, "new Name", ticketToStore);
+
+        verify(userDAO, times(1)).updateUserNameAndSaveTicket(1L, "new Name", ticketToStore);
```

Comment:
just reminder that if you want verify that method called once you just could write verify(userDao).update...